storage/range_command.go

```diff
@@ -266,17 +268,17 @@ func (r *Range) EndTransaction(batch engine.Engine, ms *engine.MVCCStats, args *
 
 		// Take max of requested epoch and existing epoch. The requester
 		// may have incremented the epoch on retries.
-		if reply.Txn.Epoch < args.Txn.Epoch {
-			reply.Txn.Epoch = args.Txn.Epoch
+		if reply.Txn.Epoch < txn.Epoch {
+			reply.Txn.Epoch = txn.Epoch
 		}
 		// Take max of requested priority and existing priority. This isn't
 		// terribly useful, but we do it for completeness.
-		if reply.Txn.Priority < args.Txn.Priority {
-			reply.Txn.Priority = args.Txn.Priority
+		if reply.Txn.Priority < txn.Priority {
+			reply.Txn.Priority = txn.Priority
 		}
 	} else {
 		// The transaction doesn't exist yet on disk; use the supplied version.
-		reply.Txn = args.Txn
+		reply.Txn = &txn
```

Comment:
Yeah, but It's a bit more of a footgun; a future refactor could introduce a regression and since we do not test for this I'd rather leave it as-is for now.